app/assets/javascripts/discourse/scripts/async-stylesheets.js

```diff
@@ -0,0 +1,19 @@
+// This script is inlined in `_discourse_stylesheet.html.erb
+const links = document.getElementsByClassName("async-css-loading");
+
+const processEvent = function (element, eventListenerCallback) {
+  element.media = element.dataset.media;
```

Comment:
If I'm understanding correctly, this will flip the `media` attribute of the stylesheet as soon as its loaded (or errored)? That means, depending on network conditions and cache, it could flip before the app boots, midway through boot, or after boot.

I think that's quite likely to lead to some weird race conditions, which will be hard to debug.

How about deferring this change until the app is completely booted? i.e. put it in the `_displayApp()` function. That way, there will be no race conditions. We can be very clear: stylesheets are not loaded until after the initial app boot/render.

Reply:
Good point, I have made that change too.